File: FitoGrano/CS/Classes/clsForSQL.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using Microsoft.VisualBasic;
using System.Xml.Linq;
using System.Windows.Forms;
// End of VB project level imports

using FitoGrano;

namespace FitoGrano
{
	public class clsForSQL
	{
		// проверка передаваемых значений в базу
		public static string InDataBase(object iValue, string iDataType = "varchar")
		{
			// If IsNumeric(iValue) = False Then
			if (ReferenceEquals(iValue, null))
			{
				return "NULL";
			}
			if (Information.IsDBNull(iValue))
			{
				return "NULL";
			}
			if (iValue.ToString() == "NULL")
			{
				return "NULL";
			}
			if (Strings.Trim(System.Convert.ToString(iValue)) == "")
			{
				return "NULL";
			}
			//End If
			if (iDataType == "varchar")
			{
				iValue = Strings.Replace(System.Convert.ToString(iValue), "'", "''");
				return "'" + Strings.Trim(System.Convert.ToString(iValue)) + "'";
			}
			if (iDataType == "int")
			{
				return Strings.Trim(System.Convert.ToString(iValue));
			}
			if (iDataType == "DateTime")
			{
				DateTime d = System.Convert.ToDateTime(iValue);
				return "'" + d.ToShortDateString() + "'";
			}
		}
		// проверка значений из бызы
		public static dynamic OutDataBase(object iValue)
		{
			if (Information.IsDBNull(iValue))
			{
				return null;
			}
			else
			{
				return iValue;
			}
		}
		// замена аппострофов
		private string CheckingApostrophe(string iString)
		{
			string iChar = ""; // проверяемый символ
			string iText = ""; // проверяемый текст
			for (var i = 1; i <= iString.Length; i++)
			{
				iChar = Microsoft.VisualBasic.Strings.Mid(iString, System.Convert.ToInt32(i), 1);
				if (iChar == "'")
				{
					iChar = "'''";
				}
				iText += iChar;
			}
			return iText;
		}
		
		
		
	}
	
}

File: FitoGrano/Classes/clsForSQL.vb
Public Class clsForSQL
    ' проверка передаваемых значений в базу
    Public Shared Function InDataBase(ByVal iValue As Object, Optional ByVal iDataType As String = "varchar") As String
        ' If IsNumeric(iValue) = False Then
        If iValue Is Nothing Then Return "NULL"
        If IsDBNull(iValue) Then Return "NULL"
        If iValue.ToString = "NULL" Then Return "NULL"
        If Trim(iValue) = "" Then Return "NULL"
        'End If
        If iDataType = "varchar" Then
            iValue = Replace(iValue, "'", "''")
            Return "'" & Trim(iValue) & "'"
        End If
        If iDataType = "int" Then Return Trim(iValue)
        If iDataType = "DateTime" Then
            Dim d As Date = iValue
            Return "'" & d.ToShortDateString & "'"
        End If
    End Function
    ' проверка значений из бызы
    Public Shared Function OutDataBase(ByVal iValue As Object)
        If IsDBNull(iValue) Then
            Return Nothing
        Else
            Return iValue
        End If
    End Function
    ' замена аппострофов
    Private Function CheckingApostrophe(ByVal iString As String) As String
        Dim iChar As String ' проверяемый символ
        Dim iText As String  ' проверяемый текст
        For i = 1 To iString.Length
            iChar = Microsoft.VisualBasic.Mid(iString, i, 1)
            If iChar = "'" Then
                iChar = "'''"
            End If
            iText &= iChar
        Next
        Return iText
    End Function



End Class
